src/main/java/com/mitchej123/hodgepodge/SimulationDistanceHelper.java

```diff
@@ -0,0 +1,332 @@
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Check if a chunk should get processed
+     */
+    public boolean shouldProcessTick(int x, int z) {
+        long key = ChunkCoordIntPair.chunkXZ2Int(x, z);
+        return cacheShouldProcessTick.computeIfAbsent(key, dummy -> {
+            if (closeToPlayer(x, z)) {
+                return true;
+            }
+            if (noTickChunks.containsKey(key)) {
+                return false;
+            }
+            return forcedChunksMap.contains(key);
+        });
+    }
+
+    private LongOpenHashSet getPlayerChunksForPos(ChunkCoordIntPair pos) {
+        int simulationDistance = getSimulationDistance();
+        LongOpenHashSet chunks = new LongOpenHashSet();
+        for (int x = -simulationDistance; x <= simulationDistance; x++) {
+            for (int z = -simulationDistance; z <= simulationDistance; z++) {
+                chunks.add(ChunkCoordIntPair.chunkXZ2Int(pos.chunkXPos + x, pos.chunkZPos + z));
+            }
+        }
+        return chunks;
+    }
+
+    private void mergeNoTickChunkChanges() {
+        synchronized (noTickChunksChanges) {
+            for (LongBooleanPair update : noTickChunksChanges) {
+                long key = update.keyLong();
+                boolean prevent = update.valueBoolean();
+                byte value = noTickChunks.getOrDefault(key, (byte) 0);
+                value += (byte) (prevent ? 1 : -1);
+                if (value > 0) {
+                    noTickChunks.put(key, value);
+                } else {
+                    noTickChunks.remove(key);
+                }
+            }
+            noTickChunksChanges.clear();
+        }
+    }
+
+    private void checkForAddedChunks(LongOpenHashSet forcedChunksOld) {
+        LongOpenHashSet added = new LongOpenHashSet();
+        int simulationDistance = getSimulationDistance();
+
+        // New forced chunks?
+        for (long chunk : forcedChunksMap) {
+            if (!forcedChunksOld.contains(chunk)) {
+                added.add(chunk);
+            }
+        }
+
+        // New players, or moved players?
+        for (EntityPlayer player : world.playerEntities) {
+            if (player.getEntityWorld() != world) {
+                continue;
+            }
+            ChunkCoordIntPair playerPos = new ChunkCoordIntPair((int) player.posX >> 4, (int) player.posZ >> 4);
+            ChunkCoordIntPair playerPosOld = playerPosOldMap.getOrDefault(player, null);
+            if (playerPosOld == null || simulationDistance != simulationDistanceOld) {
+                LongOpenHashSet chunksNewPos = getPlayerChunksForPos(playerPos);
+                added.addAll(chunksNewPos);
+            } else if (!playerPos.equals(playerPosOld)) {
+                LongOpenHashSet chunksNewPos = getPlayerChunksForPos(playerPos);
+                LongOpenHashSet chunksOldPos = getPlayerChunksForPos(playerPosOld);
+                for (long pos : chunksNewPos) {
```

Comment:
This is probably doing long boxing/unboxing, use .forEach() or a manual for loop

Reply:
https://fastutil.di.unimi.it/docs/it/unimi/dsi/fastutil/longs/LongIterator.html

Reply:
Thanks, pushed an update